WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -323,16 +325,37 @@ import WordPressFlux
         }
     }
 
+
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
 
         syncIfAppropriate()
 
+        debugCheckIfNeedToDisplaySelectInterests()
+
+    }
+
+    private func debugCheckIfNeedToDisplaySelectInterests() {
         // TODO: Remove this
         if FeatureFlag.readerImprovementsPhase2.enabled {
-            DispatchQueue.main.asyncAfter(deadline: .now() + .milliseconds(500)) {
-                let controller = ReaderSelectInterestsViewController()
-                self.navigationController?.present(controller, animated: true)
+            interestsCoordinator.shouldDisplay { shouldDisplay in
+                if shouldDisplay {
+                    let controller = ReaderSelectInterestsViewController()
+                    self.navigationController?.present(controller, animated: true)
+                }
+                self.interestsCoordinator.markAsSeen()
+            }
+        }
+    }
+
+    override func becomeFirstResponder() -> Bool {
+        return true
+    }
+
+    override func motionEnded(_ motion: UIEvent.EventSubtype, with event: UIEvent?) {
+        if FeatureFlag.readerImprovementsPhase2.enabled {
```

Comment:
What is this function all about? also, can you combine the if statements?

Reply:
It's just a debug method to help with testing. I am removing it before committing to develop. Without having a way to easily reset the has seen key you would need to completely delete the app and reinstall to test each flow.

Reply:
I've removed it: https://github.com/wordpress-mobile/WordPress-iOS/commit/4014d4b173d09ea2fde2a80c5f3454c929fb4988